### ISSUE TITLE
Go package: support file input

### ISSUE THREAD

**Issue by beatcracker** [TomWright/dasel#187]
**Is your feature request related to a problem? Please describe.**
I'm looking into using [dasel package](https://daseldocs.tomwright.me/use-as-a-go-package) in my tool. As far as I can see you're supposed to unmarshal input file yourself and then use `rootNode := dasel.New(data)` to work with the data.

**Describe the solution you'd like**
Since `dasel` command already knows how to read all supported file types, it would be nice to have something like `dasel.NewFromFile(path, fileType)`.

**Describe alternatives you've considered**
Basically re-implementing everything in the [./internal](https://github.com/TomWright/dasel/blob/master/internal) folder.

**Additional context**
I need to modify various file formats in my tool and `dasel` fits the bill perfectly. With this addition it would be much easier to use `dasel` as a package.

**Comment by TomWright** [comment]
This sounds reasonable. I'd be happy to do it at some point or if you wanted to submit a PR it would be appreciated!

**Comment by beatcracker** [comment]
I'd love to do a PR, but I'm still making baby steps in Go, so I'd need guidance on this.
